### PR TITLE
Update registry k8s.gcr.io -> registry.k8s.io

### PR DESCRIPTION
**Pull request by yashsingh74**
This PR updates the registry from `k8s.gcr.io` to `registry.k8s.io`

Issue: https://github.com/kubernetes/k8s.io/issues/4738

```release-note
Update registry k8s.gcr.io -> registry.k8s.io
```

### CODE REVIEW

File: release-tools/SIDECAR_RELEASE_PROCESS.md [hunk 1]
```diff
@@ -100,9 +100,9 @@ naming convention `<hostpath-deployment-version>-on-<kubernetes-version>`.
 1. If release was a new major/minor version, create a new `release-<minor>`
    branch at that commit.
 1. Check [image build status](https://k8s-testgrid.appspot.com/sig-storage-image-build).
-1. Promote images from k8s-staging-sig-storage to k8s.gcr.io/sig-storage. From
+1. Promote images from k8s-staging-sig-storage to registry.k8s.io/sig-storage. From
```
Comment (msau42) [inline release-tools/SIDECAR_RELEASE_PROCESS.md:103]: Files under release-tools are imported by subtree and should not be directly changed. Please update https://github.com/kubernetes-csi/csi-release-tools first.

Comment (yashsingh74) [reply release-tools/SIDECAR_RELEASE_PROCESS.md:103]: Requested for the changes in the suggested repo.

**Comment by sunnylovestiramisu** [comment]
Can you rebase?

**Pull request merged**